Index: src/main/webapp/resources/js/mealDatatables.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/resources/js/mealDatatables.js	(date 1525105649000)
+++ src/main/webapp/resources/js/mealDatatables.js	(date 1525105658888)
@@ -10,6 +10,11 @@
     });
 }
 
+function clearFilter() {
+    $("#filter")[0].reset();
+    $.get(ajaxUrl, updateTableByData);
+}
+
 $(function () {
     datatableApi = $("#datatable").DataTable({
         "paging": false,
Index: src/main/webapp/WEB-INF/jsp/meals.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/meals.jsp	(date 1525105649000)
+++ src/main/webapp/WEB-INF/jsp/meals.jsp	(date 1525105700175)
@@ -47,6 +47,10 @@
                         </form>
                     </div>
                     <div class="card-footer text-right">
+                        <button class="btn btn-danger" onclick="clearFilter()">
+                            <span class="fa fa-remove"></span>
+                            <spring:message code="common.cancel"/>
+                        </button>
                         <button class="btn btn-primary" onclick="updateTable()">
                             <span class="fa fa-filter"></span>
                             <spring:message code="meal.filter"/>
