Index: src/main/webapp/WEB-INF/jsp/users.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/users.jsp	(revision 4638c217824877b65b5dc933ffc78f283f43178f)
+++ src/main/webapp/WEB-INF/jsp/users.jsp	(date 1525175106053)
@@ -10,7 +10,7 @@
 <script type="text/javascript" src="resources/js/userDatatables.js" defer></script>
 <jsp:include page="fragments/bodyHeader.jsp"/>
 
-<div class="jumbotron">
+<div class="jumbotron pt-4">
     <div class="container">
         <h3><spring:message code="user.title"/></h3>
         <br/>
@@ -79,11 +79,11 @@
             </div>
             <div class="modal-footer">
                 <button type="button" class="btn btn-secondary" data-dismiss="modal">
-                    <span class="fa fa-close" aria-hidden="true"></span>
+                    <span class="fa fa-close"></span>
                     <spring:message code="common.cancel"/>
                 </button>
                 <button type="button" class="btn btn-primary" onclick="save()">
-                    <span class="fa fa-check" aria-hidden="true"></span>
+                    <span class="fa fa-check"></span>
                     <spring:message code="common.save"/>
                 </button>
             </div>
Index: config/messages/app_ru.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- config/messages/app_ru.properties	(revision 4638c217824877b65b5dc933ffc78f283f43178f)
+++ config/messages/app_ru.properties	(date 1525175092267)
@@ -30,5 +30,4 @@
 common.delete=Удалить
 common.update=Обновить
 common.save=Сохранить
-common.cancel=Отмена
-common.actions=Выполнить
\ No newline at end of file
+common.cancel=Отменить
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(revision 4638c217824877b65b5dc933ffc78f283f43178f)
+++ src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(date 1525175095482)
@@ -5,7 +5,7 @@
     <div class="container">
         <a href="meals" class="navbar-brand"><img src="resources/images/icon-meal.png"> <spring:message code="app.title"/></a>
         <form class="form-inline my-2">
-            <a class="btn btn-info mr-2" href="users"><spring:message code="user.title"/></a>
+            <a class="btn btn-info mr-1" href="users"><spring:message code="user.title"/></a>
             <a class="btn btn-primary" href="">
                 <span class="fa fa-sign-in"></span>
             </a>
