Index: src/main/webapp/WEB-INF/jsp/login.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/login.jsp	(date 1525173412428)
+++ src/main/webapp/WEB-INF/jsp/login.jsp	(date 1525173412428)
@@ -0,0 +1,82 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
+
+<html>
+<jsp:include page="fragments/headTag.jsp"/>
+<body>
+<nav class="navbar navbar-dark bg-dark">
+    <div class="container">
+        <div class="navbar-brand"><img src="resources/images/icon-meal.png"> <spring:message code="app.title"/></div>
+        <form class="form-inline my-2" action="spring_security_check" method="post">
+            <input class="form-control mr-1" type="text" placeholder="Email" name="username">
+            <input class="form-control mr-1" type="password" placeholder="Password" name="password">
+            <button class="btn btn-success" type="submit">
+                <span class="fa fa-sign-in"></span>
+            </button>
+        </form>
+    </div>
+</nav>
+
+<div class="jumbotron pt-0">
+    <div class="container">
+        <c:if test="${param.error}">
+            <div class="error">${sessionScope["SPRING_SECURITY_LAST_EXCEPTION"].message}</div>
+        </c:if>
+        <c:if test="${not empty param.message}">
+            <div class="message"><spring:message code="${param.message}"/></div>
+        </c:if>
+        <br/>
+        <p>
+            <button type="submit" class="btn btn-lg btn-primary" onclick="setCredentials('user@yandex.ru', 'password')">
+                <spring:message code="app.login"/> User
+            </button>
+            <button type="submit" class="btn btn-lg btn-primary" onclick="setCredentials('admin@gmail.com', 'admin')">
+                <spring:message code="app.login"/> Admin
+            </button>
+        </p>
+        <br/>
+        <p>Стек технологий: <a href="http://projects.spring.io/spring-security/">Spring Security</a>,
+            <a href="https://docs.spring.io/spring/docs/current/spring-framework-reference/html/mvc.html">Spring MVC</a>,
+            <a href="http://projects.spring.io/spring-data-jpa/">Spring Data JPA</a>,
+            <a href="http://spring.io/blog/2014/05/07/preview-spring-security-test-method-security">Spring Security
+                Test</a>,
+            <a href="http://hibernate.org/orm/">Hibernate ORM</a>,
+            <a href="http://hibernate.org/validator/">Hibernate Validator</a>,
+            <a href="http://www.slf4j.org/">SLF4J</a>,
+            <a href="https://github.com/FasterXML/jackson">Json Jackson</a>,
+            <a href="http://ru.wikipedia.org/wiki/JSP">JSP</a>,
+            <a href="http://en.wikipedia.org/wiki/JavaServer_Pages_Standard_Tag_Library">JSTL</a>,
+            <a href="http://tomcat.apache.org/">Apache Tomcat</a>,
+            <a href="http://www.webjars.org/">WebJars</a>,
+            <a href="http://datatables.net/">DataTables plugin</a>,
+            <a href="http://ehcache.org">EHCACHE</a>,
+            <a href="http://www.postgresql.org/">PostgreSQL</a>,
+            <a href="http://junit.org/">JUnit</a>,
+            <a href="http://hamcrest.org/JavaHamcrest/">Hamcrest</a>,
+            <a href="http://jquery.com/">jQuery</a>,
+            <a href="http://ned.im/noty/">jQuery notification</a>,
+            <a href="http://getbootstrap.com/">Bootstrap</a>.</p>
+    </div>
+</div>
+<div class="container">
+    <div class="lead">
+        &nbsp;&nbsp;&nbsp;<a href="https://github.com/JavaOPs/topjava">Java Enterprise проект</a> с
+        регистрацией/авторизацией и интерфейсом на основе ролей (USER, ADMIN).
+        Администратор может создавать/редактировать/удалять пользователей, а пользователи - управлять своим
+        профилем и данными (день, еда, калории) через UI (по AJAX) и по REST интерфейсу с базовой авторизацией.
+        Возможна фильтрация данных по датам и времени, при этом цвет записи таблицы еды зависит от того, превышает ли
+        сумма
+        калорий за день норму (редактируемый параметр в профиле пользователя).
+        Весь REST интерфейс покрывается JUnit тестами, используя Spring MVC Test и Spring Security Test.
+    </div>
+</div>
+<jsp:include page="fragments/footer.jsp"/>
+<script type="text/javascript">
+    function setCredentials(username, password) {
+        $('input[name="username"]').val(username);
+        $('input[name="password"]').val(password);
+    }
+</script>
+</body>
+</html>
\ No newline at end of file
Index: src/main/webapp/test.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/test.html	(date 1525172153658)
+++ src/main/webapp/test.html	(date 1525172153658)
@@ -0,0 +1,13 @@
+<html>
+<link rel="stylesheet" href="resources/css/style.css">
+<link rel="stylesheet" href="webjars/bootstrap/4.1.0/css/bootstrap.min.css">
+<link rel="shortcut icon" href="resources/images/icon-meal.png">
+<body>
+<hr/>
+<ul>
+    <li><a href="users">User list</a></li>
+    <li><a href="meals">Meal list</a></li>
+</ul>
+<hr/>
+</body>
+</html>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/index.jsp
===================================================================
--- src/main/webapp/WEB-INF/jsp/index.jsp	(date 1525171894000)
+++ src/main/webapp/WEB-INF/jsp/index.jsp	(date 1525171894000)
@@ -1,23 +0,0 @@
-<%@ page contentType="text/html;charset=UTF-8" language="java" %>
-<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
-<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
-
-<html>
-<jsp:include page="fragments/headTag.jsp"/>
-<body>
-<jsp:include page="fragments/bodyHeader.jsp"/>
-<div class="jumbotron">
-    <div class="container">
-        <form method="post" action="users" class="form-inline">
-            <label><spring:message code="app.login"/></label>
-            <select name="userId" class="form-control mx-3">
-                <option value="100000" selected>User</option>
-                <option value="100001">Admin</option>
-            </select>
-            <button type="submit" class="btn btn-primary"><spring:message code="common.select"/></button>
-        </form>
-    </div>
-</div>
-<jsp:include page="fragments/footer.jsp"/>
-</body>
-</html>
\ No newline at end of file
Index: src/main/webapp/resources/css/style.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/resources/css/style.css	(date 1525171894000)
+++ src/main/webapp/resources/css/style.css	(date 1525171996337)
@@ -14,6 +14,24 @@
     opacity: 0.3;
 }
 
+.error, .message {
+    padding: 10px;
+    border-radius: 4px;
+    font-size: 16px;
+}
+
+.error {
+    color: #a94442;
+    background-color: #f2dede;
+    border: 1px solid #ebccd1;
+}
+
+.message {
+    color: #2f9635;
+    background-color: #c6fbc2;
+    border: 1px solid #9feba6;
+}
+
 /*https://getbootstrap.com/docs/4.0/examples/sticky-footer/sticky-footer.css*/
 html {
     position: relative;
Index: src/main/resources/spring/spring-mvc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/spring/spring-mvc.xml	(date 1525171894000)
+++ src/main/resources/spring/spring-mvc.xml	(date 1525171952969)
@@ -1,12 +1,15 @@
-<beans xmlns="http://www.springframework.org/schema/beans"
-       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<beans xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns:context="http://www.springframework.org/schema/context"
-       xmlns:p="http://www.springframework.org/schema/p" xmlns:mvc="http://www.springframework.org/schema/mvc"
+       xmlns:p="http://www.springframework.org/schema/p"
+       xmlns:mvc="http://www.springframework.org/schema/mvc" xmlns="http://www.springframework.org/schema/beans"
        xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
        http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd http://www.springframework.org/schema/mvc http://www.springframework.org/schema/mvc/spring-mvc.xsd">
 
     <bean class="ru.javawebinar.topjava.web.json.JacksonObjectMapper" id="objectMapper" factory-method="getMapper"/>
 
+    <!-- serve static resources (*.html, ...) from src/main/webapp/ -->
+    <mvc:default-servlet-handler/>
+
     <mvc:annotation-driven conversion-service="conversionService">
         <mvc:message-converters>
             <bean class="org.springframework.http.converter.json.MappingJackson2HttpMessageConverter">
Index: src/main/java/ru/javawebinar/topjava/web/RootController.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/ru/javawebinar/topjava/web/RootController.java	(date 1525171894000)
+++ src/main/java/ru/javawebinar/topjava/web/RootController.java	(date 1525171952962)
@@ -4,13 +4,10 @@
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
 import ru.javawebinar.topjava.AuthorizedUser;
 import ru.javawebinar.topjava.service.MealService;
 import ru.javawebinar.topjava.util.MealsUtil;
 
-import javax.servlet.http.HttpServletRequest;
-
 @Controller
 public class RootController {
 
@@ -19,7 +16,7 @@
 
     @GetMapping("/")
     public String root() {
-        return "index";
+        return "redirect:meals";
     }
 
     @GetMapping("/users")
@@ -27,11 +24,9 @@
         return "users";
     }
 
-    @PostMapping("/users")
-    public String setUser(HttpServletRequest request) {
-        int userId = Integer.valueOf(request.getParameter("userId"));
-        AuthorizedUser.setId(userId);
-        return "redirect:meals";
+    @GetMapping(value = "/login")
+    public String login() {
+        return "login";
     }
 
     @GetMapping("/meals")
Index: src/main/resources/spring/spring-security.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/spring/spring-security.xml	(date 1525171894000)
+++ src/main/resources/spring/spring-security.xml	(date 1525171952978)
@@ -17,10 +17,15 @@
     <beans:bean name="noopEncoder" class="org.springframework.security.crypto.password.NoOpPasswordEncoder"/>
 
     <http>
+        <intercept-url pattern="/login" access="isAnonymous()"/>
+
         <intercept-url pattern="/**/admin/**" access="hasRole('ROLE_ADMIN')"/>
         <intercept-url pattern="/**" access="isAuthenticated()"/>
-        <form-login/>
-        <logout/>
+
+        <form-login login-page="/login" default-target-url="/meals"
+                    authentication-failure-url="/login?error=true"
+                    login-processing-url="/spring_security_check"/>
+        <logout logout-success-url="/login"/>
         <csrf disabled="true"/>
     </http>
 
